### PR TITLE
tidy up internal link transformer

### PR DESCRIPTION
**Pull request by jroebu14**
Related to: #4973

**Overall change:** 
Refactors the function that transforms internal links XML into something that a subsequent function which transforms all links into Optimo format can handle. The changes include breaking bigger functions up into smaller functions where possible and making them return new values instead of mutating values passed in.

**Code changes:**
- Refactors `internalLinks` preprocessor rule within `convertToOptimoBlocks`

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added labels to this PR for the relevant pod(s) affected by these changes
- [x] I have assigned this PR to the Simorgh project

**Testing:**
- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [ ] This PR requires manual testing

### CODE REVIEW

File: src/app/lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks/blocks/internalLinks/index.js [hunk 1]
```diff
@@ -21,82 +20,76 @@ to the same schema used for external links
...
+  `
+<link>
+    <caption>${text}</caption>
+    <altText>${text}</altText>
+    <url href="${url}" platform="highweb"/>
+    <url href="${url}" platform="enhancedmobile"/>
+</link>
+`.replace(whiteSpaceRegex, '');
+
+const replaceChevrons = text => text.replace('<', '%3C').replace('>', '%3E');
```
Comment (ryanmccombe) [inline src/app/lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks/blocks/internalLinks/index.js]: Needs changed to this following the update to fix the chevron issue: https://github.com/bbc/simorgh/pull/5143/files

(also has updated unit tests in that PR)

```suggestion
const replaceChevrons = text => text.replace('<', '&lt;').replace('>', '&gt;');
```

**Review by ryanmccombe — APPROVED** [review]
Looks good - thanks :)

**Review by ryanmccombe — APPROVED** [review]

**Review by rebeccamcginn — APPROVED** [review]
Looking snazzy

**Comment by jroebu14** [comment]
> Looking snazzy

Best code review comment I've ever had 😄Thanks!

**Review by ryanmccombe — CHANGES_REQUESTED** [review]
Live issue - do not merge yet

**Comment by ryanmccombe** [comment]
This PR needs integrated: https://github.com/bbc/simorgh/pull/5171

**Review by rebeccamcginn — APPROVED** [review]

**Review by simonsinclair — APPROVED** [review]
Nice! We're adding some preprocessors for the PGL pages, which currently mutate the block model. This work provides some great examples of how we could refactor them to avoid doing that.

**Review by ryanmccombe — APPROVED** [review]

**Comment by jroebu14** [comment]
> Nice! We're adding some preprocessors for the PGL pages, which currently mutate the block model. This work provides some great examples of how we could refactor them to avoid doing that.

@simonsinclair feel free to give me a shout if you would like to pair or need some pointers when doing that!

**Pull request merged**